alacritty_terminal/src/term/mod.rs

```diff
@@ -1875,8 +1890,23 @@ impl<T: EventListener> Handler for Term<T> {
                     let template = Cell { bg: template.bg, ..Cell::default() };
                     self.grid.clear_viewport(template);
                 }
+
+                self.selection =
+                    self.selection.take().filter(|s| !s.intersects_range(0..=last_line));
+            },
+            ansi::ClearMode::Saved => {
+                let last_scrollback_line = self.grid.len() - 1;
+                if last_scrollback_line == last_line {
+                    return;
+                }
```

Comment:
Imo the early return here is much, much more error prone and complex than having a `_ => ()` match arm.

You can also just do `ClearMode::Saved => ()`, if you're concerned about adding new variants.